test_util/launch.py

```diff
@@ -1,19 +1,136 @@
-"""
+"""DC/OS Launch
 
 Usage:
-dcos-launch create [--wait] [--dump-info=cluster_info.json] <config=config.yaml>
-dcos-launch wait <path=cluster_info.json>
-dcos-launch describe <path=cluster_info.json>
-dcos-launch delete <path=cluster_info.json>
-"""
+  dcos-launch create [--info-path=<path>] [--config-path=<path>]
+  dcos-launch wait [--info-path=<path>]
+  dcos-launch (describe|delete) [--info-path=<path>]
 
+Commands:
+  create    reads the file given by --config-path, creates the cluster described therein
+            and finally dumps a JSON file to the path given in --info-path which can then
+            be used with the wait, describe, and delete calls
+  wait      Block until the cluster is up and running
+  describe  Return additional information about the composition of the cluster
+  delete    Destroying the provided cluster deployment
+
+Options:
+  --config-path=<path>  Path for config to create cluster from [default: config.yaml].
+  --info-path=<path>    JSON file output by create and consumed by wait, describe,
+                        and delete [default: cluster_info.json].
+"""
+import abc
+import json
+import os
+import sys
+import yaml
 from docopt import docopt
+from pkgpanda.util import load_json, load_string, write_json
+
+from test_util.aws import BotoWrapper, DcosCfSimple
+
+
+class AbstractLauncher(metaclass=abc.ABCMeta):
+    def create(self):
+        raise NotImplementedError()
+
+    def wait(self):
+        raise NotImplementedError()
+
+    def describe(self):
+        raise NotImplementedError()
+
+    def delete(self):
+        raise NotImplementedError()
+
+
+class AwsCloudformationLauncher(AbstractLauncher):
+    def __init__(self, config):
```

Comment:
I think we need to workout whether the config is required for all options / operations or not. This code as it is currently is suggesting that yes, the config.yaml (because it contains the AWS Credentials) is necessary for any / all usage of AwsCloudformation stack, so the CLI needs to always provide / specify it.

I'm fine with that as a design, but we probably need to adapt the CLI docopt string to it.

Another approach we could go with is "create" is the only thing that needs the config, then _everything_ (including credentials) which is critical to continued operation is stored in the cluster_info, and only the cluster_info is needed at that point to create a Launcher.